lib/cocoapods/command/repo/push.rb

```diff
@@ -205,16 +206,7 @@ def push_repo
         # @return [Pathname] The directory of the repository.
         #
         def repo_dir
-          specs_dir = Pathname.new(File.join(config.repos_dir, @repo, 'Specs'))
-          dir = config.repos_dir + @repo
-          if specs_dir.exist?
-            dir = specs_dir
-          elsif dir.exist?
-            dir
-          else
-            raise Informative, "`#{@repo}` repo not found either in #{specs_dir} or #{dir}"
-          end
-          dir
+          @source.specs_dir
```

Comment:
`Source.specs_dir` is currently a private method in Core, so that would also need to be made public, but way nicer than recalculating the path each time.